src/web/mjs/connectors/MangaReaderTo.mjs

```diff
@@ -0,0 +1,121 @@
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, this.queryMangas);
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title: element.title.trim()
+            };
+        });
+    }
+
+    async _getChapters(manga) {
+        let uri = new URL(manga.id, this.url);
+        let request = new Request(uri, this.requestOptions);
+        let data = await this.fetchDOM(request, this.queryChapters);
+        return data.map(element => {
+            const link = this.getRootRelativeOrAbsoluteLink(element, this.url);
+            const title = element.title.replace(/:(.*)/gi, '');
+            const lang = link.match(/(\/en\/)|(\/ja\/)/gi);
+            const language = lang ? lang[0].replace(/\//gm, '').toUpperCase() : '';
+            return {
+                id: link,
+                title: title.replace(manga.title, '').trim() + ` ${language}` || manga.title,
+                language
+            };
+        });
+    }
+
+    async _getPages(chapter) {
+        const uri = new URL(chapter.id, this.url);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, this.queryPages);
+        const readingId = data[0].getAttribute('data-reading-id');
+
+        const script = await this._getImages(readingId);
+        const images = await Engine.Request.fetchUI(request, script, 60000, true);
+
+        return images.map(page => this.createConnectorURI(page));
+    }
+
+    async _getImages(readingId) {
+        // https://mangareader.to/ajax/image/list/chap/545260?mode=vertical&quality=high&hozPageSize=1
+        const uri = new URL(`ajax/image/list/chap/${readingId}`, this.url);
+        uri.searchParams.set('quality', 'high');
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request, 3);
+
+        const dom = this.createDOM(data.html);
+        const images = Array.from(dom.querySelectorAll('.iv-card'));
+
+        // Example: https://c-1.mreadercdn.ru/_v2/1/0dcb8f9eaacfd940603bd75c7c152919c72e45517dcfb1087df215e3be94206cfdf45f64815888ea0749af4c0ae5636fabea0abab8c2e938ab3ad7367e9bfa52/9d/32/9d32bd84883afc41e54348e396c2f99a/9d32bd84883afc41e54348e396c2f99a_1200.jpeg?t=4b419e2c814268686ff05d2c25965be9&amp;ttl=1642926021
+        const imageData = JSON.stringify(images.map(image => image.getAttribute('data-url')));
+
+        return `
+            new Promise((resolve, reject) => {
+                (async () => {
+                    try {
+                        let images = [];
+                        const data = ${imageData};
+                        for(const d of data) {
+                            const canvas = await imgReverser(d);
+                            const uri = canvas.toDataURL('image/jpeg')
```

Comment:
Aren't data URLs limited to max. 2MB or something lime that in Chrome?
Use the descrambling format from settings that were configured by the user:
`Engine.Settings.recompressionFormat.value`
and
`parseFloat(Engine.Settings.recompressionQuality.value)/100`